cmd/neofs-node/attributes.go

```diff
@@ -11,4 +15,52 @@ func parseAttributes(c *cfg) {
 	}
 
 	fatalOnErr(attributes.ReadNodeAttributes(&c.cfgNodeInfo.localInfo, nodeconfig.Attributes(c.appCfg)))
+
+	// expand UN/LOCODE attribute if any found; keep user's attributes
+	// if any conflicts appear
+
+	locAttr := c.cfgNodeInfo.localInfo.LOCODE()
+	if locAttr == "" {
+		return
```

Comment:
lets changelog that storage nodes no longer obliged to set LOCODE attribute from now

Reply:
why do you think this is a _change_?

Reply:
cuz they were obliged before

Reply:
nope, they werent bcfd33d9fb1476b2555b807c476d8e951c54ace9
my bad sry

Reply:
...but now we have https://github.com/nspcc-dev/neofs-sdk-go/blob/d72d76ccb0baa0ed61b32a214ee67997661303de/netmap/node_info.go#L317-L318

finally: LOCODE attrs' absence is valid or invalid?

Reply:
> finally: LOCODE attrs' absence is valid or invalid?

need to be decided. you specified the commit, you mentioned the SDK getter and i can add some words about "well-known" attrs from the API that i am not sure i can understand clearly: https://github.com/nspcc-dev/neofs-api/blob/1424834f743842b88e02ec6ac79c64ef2d2d3e75/netmap/types.proto#L156-L158

@roman-khimov, lets fight

P.S.

> but now we have

not now, for a year (before the refactor, and maybe even more in fact)

Reply:
#2612 requires all attributes match the DB record when LOCODE is specified. So lets adjust SDK docs and storage node code to this requirement at the moment. Btw in current PR storage node doesn't check config settings against the DB

Reply:
> Btw in current PR storage node doesn't check config settings against the DB

that is the idea. if a user wants to place some attrs, he can do it

> https://github.com/nspcc-dev/neofs-node/issues/2612 requires all attributes match the DB record when LOCODE is specified

on the IR side. but what if a user wants Chelyabinsk to be in USA and does everything he can as an admin (config)?

> So lets adjust SDK docs and storage node code to this requirement at the moment.

so what behavior do you want? cannot fix what i cannot understand

Reply:
It's a policy thing that can vary from network to network (hi, #2279), for public networks it's more like mandatory, in private ones no one cares. But until we have #2279 we better have it optional.

Reply:
ok, do we need a warning at least?

Reply:
You can add a warning.

Reply:
done